### DISCUSSION TITLE
Want "Reject all" button to reject all cookies

### DISCUSSION THREAD

**Discussion by LokiDieKatze** [orestbida/cookieconsent#259] [Q&A] [answered]
Hi,
I am not very experienced: I would like to know if it's possible for me to change the behaviour behind button "Reject all"? I don't want the button to allow necessary cookies, it should allow no cookie at all. Also, the toggle for necessary cookies in the settings should be enabled so that the user gets to choose.
If this is an option? If so, where/how can I implement those changes?
Thanks a lot !

**Comment by orestbida** [comment, marked as answer]
@LokiDieKatze,

in order to achieve a **true reject all** you need to set the `readonly` property to `false`, for each category defined inside each language.

Make sure that all defined `toggle` objects look like this:
```javascript
toggle: {
    value: 'necessary',
    enabled: false,
    readonly: false          // cookie categories with "readonly: true" are all treated as "necessary cookies". change to false
}
```

The necessary category is generally used for critical cookies (non tracking) and services which are created/executed regardless of the user's choice. I'd recommend you to rename this category to something else (or completely remove it) if you don't need it.

**Reply by LokiDieKatze** [reply]
Thank you, it was helpful!

**Comment by LokiDieKatze** [comment]
Is it possible with cookieconsent that the user changes its preferences afterwards through a "cookie preference" page in the website? Or is the only way the cookie banner?

**Reply by tillsanders** [reply]
Yes, you can always show the settings modal by adding a link like this:

```html
<button type="button" data-cc="c-settings">Show cookie settings</button>
```

More on that here: https://github.com/orestbida/cookieconsent#available-data-cc-actions

> Edit: There is no way to have cookieconsent add a real page / URL though. But you can ...

Or you can build your own UI and connect it to the API methods: https://github.com/orestbida/cookieconsent#api-methods
